x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -311,8 +308,35 @@ func (k Keeper) proofRequirementForClaim(ctx sdk.Context, claim *prooftypes.Clai
 		"claim does not require proof due to compute units (%d) being less than the threshold (%d) and random sample (%.2f) being greater than probability (%.2f)",
 		numClaimComputeUnits,
 		proofParams.GetProofRequirementThreshold(),
-		randFloat,
+		proofRequirementCheckValue,
 		proofParams.GetProofRequestProbability(),
 	))
 	return requirementReason, nil
 }
+
+// getEarliestSupplierProofCommitBlockHash returns the block hash of the earliest
+// block at which a claim might have its proof committed.
+func (k Keeper) getEarliestSupplierProofCommitBlockHash(
+	ctx context.Context,
+	claim *prooftypes.Claim,
+) (blockHash []byte, err error) {
+	sharedParams, err := k.sharedQuerier.GetParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	sessionEndHeight := claim.GetSessionHeader().GetSessionEndBlockHeight()
+	supplierOperatorAddress := claim.GetSupplierOperatorAddress()
+
+	proofWindowOpenHeight := shared.GetProofWindowOpenHeight(sharedParams, sessionEndHeight)
+	proofWindowOpenBlockHash := k.sessionKeeper.GetBlockHash(ctx, proofWindowOpenHeight)
+
+	earliestSupplierProofCommitHeight := shared.GetEarliestSupplierProofCommitHeight(
```

Comment:
Please add a TODO_TECHDEBT next to `GetEarliestSupplierProofCommitHeight` to take in just `sharedParams & claim`

Reply:
If we only pass `sharedParams` and `claim`, this function would no longer be static, as it needs to fetch the a block hash (which is done differently on-chain and off-chain).

Reply:
Please add a `TODO_TECHDEBT` next to `GetEarliestSupplierProofCommitHeight` that says:

```
TODO_TECHDEBT: Update the method header of this function to accept (sharedParams, Claim, BlockHash).
After doing so, please review all calling sites and simplify them accordingly.
```